packages/ember-data/lib/serializers/embedded-records-mixin.js

```diff
@@ -299,13 +299,17 @@ var EmbeddedRecordsMixin = Ember.Mixin.create({
     }
     var includeIds = this.hasSerializeIdsOption(attr);
     var includeRecords = this.hasSerializeRecordsOption(attr);
-    var key;
+    var key, hasMany;
     if (includeIds) {
       key = this.keyForRelationship(attr, relationship.kind, 'serialize');
       json[key] = snapshot.hasMany(attr, { ids: true });
     } else if (includeRecords) {
       key = this.keyForAttribute(attr, 'serialize');
-      json[key] = snapshot.hasMany(attr).map(function(embeddedSnapshot) {
+      hasMany = snapshot.hasMany(attr);
+
+      Ember.warn("The relationship '" + key + "' is unknown for '" + snapshot.modelName + "' with id '" + snapshot.id + "'. Please include it in your original payload.", Ember.typeOf(hasMany) !== 'undefined');
```

Comment:
should probably mention that it's an embedded relationship, and say undefined instead of unknown